reconfigure Test sequence for the spi drivers on stm32

FRASTM opened the pull request:
This patch introduces a new `OVERLAY_CONFIG="overlay-stm32-spi-dma.conf" `for the stm32 mcus when testing the SPi with DMA transfer

The tests/drivers/spi/spi_loopback is running with SPI interrupt mode on most of the stm32 boards.
Additionally some selected target boards where a DMA is configured, can _also_ run the tests/drivers/spi/spi_loopback  with DMA transfer.

Note that The Hardware fixture "spi_loopback" is required with MOSI and MISO pins of the SPI instance are connected on the target board.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
tests: drivers: spi loopback testcase yaml for stm32 dma

This commit adds two stm32 config to execute the testcase
when the SPI is using interrupt mode for transfer.
when the SPI is using DMA for transfer (not Interrupt nor ASYNC mode).
to run the on some specific (listed) stm32 boards on SPI instance.
Note the hw fixture (physical connection on the board)

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
select the useful target boards to execute the tests/drivers/spi/spi_loopback 

-  in spi interrupt mode :  on stm32f3_disco nucleo_g071rb nucleo_g474re nucleo_f207zg nucleo_f429zi nucleo_wb55rgnucleo_f746zg nucleo_g474re nucleo_wb55rg
- in DMA mode  on  nucleo_g474re nucleo_f207zg nucleo_f429zi nucleo_wb55rg nucleo_f746zg nucleo_g474re nucleo_wb55rg

FRASTM commented:
configure the overlay for th nucleo_f746zg   --> map the SPI2 pins and use the DMA1

erwango commented:
@FRASTM CI issue should go away after a rebase

FRASTM commented:
rebase on 479215100d132044889750e88d7a48f7b4913232
add the test case tests/drivers/spi/soi_loopback on the nucleo_h743zi in DMA mode (testcase.yaml)

ABOSTM commented on tests/drivers/spi/spi_loopback/testcase.yaml:
```diff
@@ -17,3 +17,11 @@ tests:
+    harness_config:
+      fixture: spi_loopback
+    platform_allow: nucleo_g474re nucleo_f207zg nucleo_f429zi nucleo_wb55rg
+        nucleo_f746zg nucleo_g474re nucleo_wb55rg
```
nucleo_g474re and nucleo_wb55rg are duplicated

FRASTM replied:
remove extra board name

ABOSTM commented on tests/drivers/spi/spi_loopback/boards/nucleo_h743zi.conf:
```diff
@@ -1,7 +1,6 @@
-CONFIG_SPI_STM32_DMA=y
-CONFIG_SPI_STM32_INTERRUPT=n
+CONFIG_SPI_STM32_INTERRUPT=y
 CONFIG_SPI_ASYNC=n
```
Should you remove `CONFIG_SPI_ASYNC=n` like for other boards ?

FRASTM replied:
yes, removed

erwango commented on tests/drivers/spi/spi_loopback/boards/nucleo_f103rb.conf:
```diff
@@ -1,2 +1,3 @@
 CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_1"
 CONFIG_SPI_STM32_INTERRUPT=y
+CONFIG_SPI_LOOPBACK_MODE_LOOP=n
```
According to https://docs.zephyrproject.org/1.14.0/guides/kconfig/index.html#redundant-defaults, bool default to n.
So we should be able to remove `CONFIG_SPI_LOOPBACK_MODE_LOOP=n` from all boards.conf files

FRASTM replied:
removing `CONFIG_SPI_LOOPBACK_MODE_LOOP=n`
this could be completed with a 
```
    extra_configs:
      - CONFIG_SPI_STM32_INTERRUPT=y
```
in the testcase.yaml

erwango commented on tests/drivers/spi/spi_loopback/testcase.yaml:
```diff
@@ -23,5 +23,5 @@ tests:
-    platform_allow: nucleo_g474re nucleo_f207zg nucleo_f429zi nucleo_wb55rg
-        nucleo_f746zg nucleo_g474re nucleo_wb55rg
+    platform_allow: nucleo_g474re nucleo_f207zg nucleo_f429zi
+        nucleo_f746zg nucleo_wb55rg nucleo_h743zi
```
Please move `nucleo_wb55rg` and `nucleo_g474re` duplicated erase to first commit, or fix current commit message

FRASTM committed:
tests: drivers: spi loopback remove conf for most of the stm32 boards

The testcase is including a generic configuration conf
for the stm32 target boards when running the SPI loopback in
interrupt and DMA mode. Thus, the board specific conf file is useless.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
Rebase
Simply the testcase for stm32 boards with to special overlay passed as "extra_args:" to the testcase.yaml: 
`OVERLAY_CONFIG="overlay-stm32-spi.conf ` --> testing the interrupt mode
`OVERLAY_CONFIG="overlay-stm32-spi-dma.conf` --> testing the dma mode
All have SPI_1 instance enabled except when overwritten by a particular boards/xxx.conf file

For running locally, use : 
`$ west build -p auto -b nucleo_g474re -- -DOVERLAY_CONFIG=overlay-stm32-spi.conf tests/drivers/spi/spi_loopback/
` 
or
`$ west build -p auto -b nucleo_g474re -- -DOVERLAY_CONFIG=overlay-stm32-spi-dma.conf tests/drivers/spi/spi_loopback/
`

erwango commented on tests/drivers/spi/spi_loopback/boards/96b_carbon.conf:
```diff
@@ -1 +1,3 @@
 CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_2"
+CONFIG_SPI_STM32_INTERRUPT=y
+CONFIG_SPI_ASYNC=n
```
`CONFIG_SPI_ASYNC=n` could be removed

FRASTM replied:
removed

FRASTM commented:
rebase
add nucleo_wl55jc and nucleo_h743zi boards for testing spi loopback with dma

erwango commented on tests/drivers/spi/spi_loopback/testcase.yaml:
```diff
@@ -6,6 +6,7 @@ common:
 tests:
   drivers.spi.loopback:
     harness: ztest
+    filter: not CONFIG_SPI_STM32
```
```suggestion
    filter: not CONFIG_SOC_FAMILY_STM32
```

FRASTM commented:
filtering on  not CONFIG_SOC_FAMILY_STM32

FRASTM commented:
replace the list of stm 32 target boards in  "platform_allow " with a  `filter: CONFIG_SOC_FAMILY_STM32`

FRASTM committed:
tests: drivers: spi loopback testcase on spi2 for stm32 boards

Overwrites the overlay for testing the SPI2 when running on
nucleo_f302r8 and 96b_carbon targets.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on tests/drivers/spi/spi_loopback/boards/96b_carbon.conf:
```diff
@@ -1 +1,2 @@
 CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_2"
+CONFIG_SPI_STM32_INTERRUPT=y
```
I wonder why this is required, as any STM32 target, `overlay-stm32-spi.conf` should be applied.

FRASTM replied:
no, the `config SPI_STM32_INTERRUPT`  is set for each board upon  `depends on SPI`

erwango replied:
Right, my bad.

erwango commented on tests/drivers/spi/spi_loopback/boards/nucleo_f302r8.conf:
```diff
@@ -1 +1,2 @@
 CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_2"
+CONFIG_SPI_STM32_INTERRUPT=y
```
same here.

FRASTM replied:
no, the config SPI_STM32_INTERRUPT is set for each board upon depends on SPI

erwango reviewed (changes requested):
One last comment

Side question: do I understand correctly that spi_loopback test expects interrupt mode to be enabled ?

erwango commented:
@ABOSTM would you mind having a new look ?

ABOSTM reviewed (approved):
LGTM
